Automattic/woocommerce-payments
# Added logic to always fetch from the server if running the async job

Fixes #3150 

#### Changes proposed in this Pull Request



Additional fix for the issue mentioned in #3150. The previous fix still caused the spikes, because there was a brief moment where the cache was cleared before being re-fetched. This PR implements a force refresh which prevents the need to clear the cache first.







#### Testing instructions





Repeat the instructions from #3150:

* Using the WCPay Dev Tools, clear the account cache contents.
* Go to localhost:8083 and access the wp_options table, search for the following option_name: wcpay_account_data. Make a note of the expires time in the option_value. It should be 2.5 hours ahead of the time when the cache was cleared.
* Go to Tools > Scheduled Actions. Click the 'Pending' tab. Look for an action called wcpay_refresh_account_cache.
* The Scheduled Date for this action should be 2 hours ahead of when the cache was cleared.
* Wait a few minutes, and then run the scheduled action.
* Check the wp_options table again, make sure the expires time in the option_value is now 2.5 hours ahead of when you ran the scheduled action.
* Check that a new wcpay_refresh_account_cache action is in the pending actions section, and the scheduled date is 2 hours ahead of when you ran the scheduled action.

-------------------


**Post merge**
